verilog/analysis/flow_tree.cc

```diff
@@ -0,0 +1,147 @@
+            edges_[*edge_from_iterator].push_back(
+                *edge_to_iterator +
+                (edge_to_iterator !=
+                 current_if_block.end() - 1));  // add the possible edge.
+          }
+        }
+        ifs_.pop_back();  // the if block edges were added, ready to pop it.
+      }
+    }
+    token_index++;  // increment the token index.
+  }
+
+  // Adding edges for non-if blocks.
+  token_index = 0;
+  for (auto current_token : source_sequence_) {
+    current_token_enum = current_token.token_enum();
+    if (current_token_enum != PP_else && current_token_enum != PP_elsif) {
+      if (token_index > 0)
+        edges_[token_index - 1].push_back(
+            token_index);  // edges from a token to the one coming after it
+                           // directly.
+    } else {
+      if (token_index > 0)
+        edges_[token_index - 1].push_back(
+            edges_[token_index]
+                .back());  // edges from the last token in `ifdef/`ifndef body
+                           // to `endif from the same if block.
+    }
+    token_index++;  // increment the token index.
+  }
+
+  return absl::OkStatus();
+}
+
+// Traveses the control flow tree in a depth first manner, appending the visited
+// tokens to current_sequence_, then adding current_sequence_ to variants_ upon
+// completing.
+absl::Status FlowTree::DepthFirstSearch(int current_node_index) {
+  // skips preprocessor directives so that current_sequence_ doesn't contain
+  // any.
+  const auto& current_token = source_sequence_[current_node_index];
+  if (current_token.token_enum() != PP_Identifier &&
+      current_token.token_enum() != PP_ifndef &&
+      current_token.token_enum() != PP_ifdef &&
+      current_token.token_enum() != PP_define &&
+      current_token.token_enum() != PP_define_body &&
+      current_token.token_enum() != PP_elsif &&
+      current_token.token_enum() != PP_else &&
+      current_token.token_enum() != PP_endif)
+    current_sequence_.push_back(current_token);
+
+  // do recursive search through every possible edge.
+  for (auto next_node : edges_[current_node_index]) {
+    if (auto status = FlowTree::DepthFirstSearch(next_node); !status.ok()) {
+      std::cerr << "ERROR: DepthFirstSearch fails\n";
+      return status;
+    }
+  }
+  if (current_node_index ==
+      int(source_sequence_.size()) -
+          1) {  // if the current node is the last one, push the completed
+                // current_sequence_ to variants_.
+    variants_.push_back(current_sequence_);
+  }
+  if (current_token.token_enum() != PP_Identifier &&
+      current_token.token_enum() != PP_ifndef &&
+      current_token.token_enum() != PP_ifdef &&
+      current_token.token_enum() != PP_define &&
+      current_token.token_enum() != PP_define_body &&
+      current_token.token_enum() != PP_elsif &&
+      current_token.token_enum() != PP_else &&
+      current_token.token_enum() != PP_endif)
```

Comment:
For readability reasons, if your `if` is more than one line long, use `{ ... }` blocks

```
  if (this_thing ||
      the_other_thing) {
   do_stuff();
 }
```

(also in other places in this file)